astropy/units/photometric.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""
+This module defines magnitude zero points and related photometric quantities.
+"""
+
+
+import numpy as _numpy
+from .core import UnitBase, def_unit
+
+from ..constants import si as _si
+from . import cgs, si, astrophys
+
+
+_ns = globals()
+
+def_unit(['Bol', 'L_bol'], _si.L_bol0, namespace=_ns, prefixes=False,
+         doc="Luminosity corresponding to absolute bolometric magnitude zero")
+def_unit(['bol', 'f_bol'], _si.L_bol0 / (4 * _numpy.pi * (10.*astrophys.pc)**2),
+         namespace=_ns, prefixes=False, doc="Irradiance corresponding to "
+         "appparent bolometric magnitude zero")
+def_unit(['ABflux'], 3631e-23 * cgs.erg * cgs.cm**-2 / si.s / si.Hz,
+         namespace=_ns, prefixes=False,
+         doc="AB magnitude zero flux density.")
+def_unit(['STflux'], 3631e-12 * cgs.erg * cgs.cm**-2 / si.s / si.AA,
+         namespace=_ns, prefixes=False,
+         doc="ST magnitude zero flux density.")
+
+def_unit(['maggy', 'mgy'],
+         namespace=_ns, prefixes=False,
+         doc="Maggies - a linear flux unit that is the flux for a mag=0 object."
+             "To tie this onto a specific calibrated unit system, the "
+             "phot_zero_point equivalency should be used.")
+def_unit(['nanomaggy', 'nmgy'], 1e-9 * maggy,
+         namespace=_ns, prefixes=False,
+         doc="Nanomaggy, i.e. ABmag=22.5")
+
+
+def phot_zero_point(flux0=1.0*ABflux):
```

Comment:
Just `photometric_zero_point` or `zero_point` or `zero_point_flux`? (Latter because this is in `photometric` already) I also think this should *not* have a default; the user should work in AB magnitudes if it is known to be correct.

Reply:
I like `zero_point_flux` - good point there.

I'm less sure about the non-default... Honestly I think the only times I've seen "maggies" are in AB-based systems... but I can see your argument so I'll take it out for now.  Always easier to add it back in then take it away!